src/iceberg/json_internal.cc

```diff
@@ -529,7 +528,7 @@ Result<std::unique_ptr<PartitionSpec>> PartitionSpecFromJson(
     ICEBERG_ASSIGN_OR_RAISE(auto partition_field, PartitionFieldFromJson(field_json));
     partition_fields.push_back(std::move(*partition_field));
   }
-  return std::make_unique<PartitionSpec>(schema, spec_id, std::move(partition_fields));
+  return std::make_unique<PartitionSpec>(spec_id, std::move(partition_fields));
```

Comment:
As discussed offline, we cannot use the unbound form here. Instead, we need to use a new `PartitionSpec::Make` to find the source field of each partition field from `schema` and then verify it is acceptable by the transform object.